server/controllers/authController.js

```diff
@@ -86,108 +86,145 @@ let AuthController = function (
+        const expires = moment().add(1, 'days').format();
+        return invitationTokenRepository.add(req.body.email, token, expires, true)
+          .then(() => {
+            const emailClient = new EmailClient(emailClientTypes.NOREPLY);
+            return emailClient.send(req.body.email,
+              "Invitation to City Harvest",
+              `Hello from Mobilise,
 You have been invited to join City Harvest's Mobilise application.
 Please click the following link to begin creating your account on Mobilise.
 
 ${process.env.WEB_URL}/signup?token=${token}
 
-This link will expire in 24 hours.`)
-                    })
-            })
-            .then(() => res.status(200).json({message: "Success! Invitation has been sent."}))
-            .catch(err => res.status(500).json({message: err}));
-    };
+This link will expire in 24 hours.`);
+          })
+      })
+      .then(() => res.status(200).json({message: "Success! Invitation has been sent."}))
+      .catch(err => res.status(500).json({message: err}));
+  };
 
-    this.loginUser = function (req, res) {
-        let lastLogin;
-        let loggedInUser;
-        userRepository
-            .getByEmail(req.body.email)
-            .then(user => {
-                loggedInUser = user;
-                if (user && validatePassword(req.body.password, user.password)) {
-                    return user;
-                }
-                res.status(400).json({message: "Invalid username/password"});
-            })
-            .then(user => {
-                lastLogin = user.lastLogin;
-                const currentDate = moment();
-                return userRepository.update(user, {
-                    lastLogin: currentDate
-                });
-            })
-            .then(() => {
-                res.status(200).json({
-                    message: "Successful login!",
-                    user: {
-                        uid: loggedInUser.id,
-                        isAdmin: loggedInUser.isAdmin,
-                        lastLogin: lastLogin,
-                        token: generateToken(loggedInUser)
-                    }
-                });
-            })
-            .catch(() => res.status(500).send("An error occurred"));
-    };
+  this.loginUser = function (req, res) {
+    let lastLogin;
+    let loggedInUser;
+    userRepository
+      .getByEmail(req.body.email)
+      .then(user => {
+        loggedInUser = user;
+        if (user) {
+          const currentDate = moment();
+          if (user.lastLocked === null || user.lastLocked < currentDate) {
+            if (user.passwordRetries > 0) {
+              if (validatePassword(req.body.password, user.password)) {
+                return user;
+              }
+              const newPasswordRetries = user.passwordRetries - 1;
+              res.status(400).json({message: "Invalid Username/Password. Please try again."});
+              return userRepository.update(user, {
```

Comment:
This code may try to use res twice. Make sure that res.status(400) is the last thing that happens. You do not want to access the DB after sending your response.